makefile commands for pre-staging test environments (#350)
athenianco/athenian-webapp: docs/testing.md

> # Testing

Reviewer: I'd also add that having `gcloud` is a prerequisite.

In order to enhance our testing workflow we agreed to have an ability to dynamically setup [test pre-staging environments](https://athenianco.atlassian.net/browse/ENG-814)
so Product team can verify web-app changes manually on the staging data.

## Before running test environment

> Make sure that Travis job `branch-image` completed successfully 

Reviewer: I could not locally run the test env without setting the project beforehand, and I was not able to know the name of the `project` to set.

```suggestion
Make sure that Travis job `branch-image` completed successfully 

You might also need to set up the project running `gcloud config set project athenian-1` in advance.
```

Reviewer: Also, I'd mention here how to get the needed `TEST_SERVICE_ACCOUNT` env var.

## Preparing test environment

> To create a test environment simply run  
> ```shell script
> TEST_BRANCH="your-branch" make create-test-env
> ```
> Then port-forward services with
> ```shell script
> TEST_BRANCH="your-branch" make port-forward-test-env

Reviewer: ```suggestion
To create a test environment and port-forward services, simply run:
```shell script
TEST_BRANCH="your-branch" make prepare-test-env
```
```

Author: I did that intentionally because the last command of port-forwarding is a hanging one, so a person who runs `prepare-test-env` for the first time may just think that an execution has stuck.

But if you don't see any problems with this I will update the doc.

Reviewer: You made a good point.
An alternative would be to print a message like in the example before running the hanging command
```
$ make prepare-test-env
...
echo -e "\033[0;33mYour local :3000 and :8080 ports are being forwarded to gcloud ones."
echo -e "\033[0;33mPress Ctrl+C to stop."
```

But I don't know if the output of the hanging command will hide that notice.
Anyhow, I don't want to block this feature because of this suggestion, feel free to handle it on your own. You actually did a great improvement here. Even with docs.
